client/src/Pages/PotentialDefs.js

```diff
@@ -6,20 +6,35 @@ import { theme, ThemeProvider } from '../Libraries/ReactToolboxLibrary';
 
 import '../assets/react-toolbox/theme.css';
 
+const formatForResultList = apiResponse => apiResponse.map(record => 
+  Object.assign(record, {
+    rejectCb: function () {
+       alert("reject potential definition: " + this.term + " id: " + this.entry_id);
+       fetch('/entries/setstatus/3/id/' + this.entry_id, {method: 'POST'});
+    },
+
+    acceptCb: function () {
+       alert("reject potential definition: " + this.props.entry.term + " id: " + this.props.entry.entry_id);
+       fetch('/entries/setstatus/3/id/' + this.props.entry.entry_id, {method: 'POST'});
+    }
+  })
+)
 class PotentialDefs extends Component {
 
-    constructor() {
-        super();
-        this.state = {
-            potentials: []
-        }
-    }
+  constructor() {
+      super();
+      this.state = {
+          potentials: []
+      }
+  }
 
   componentDidMount(){
-      let { auth } = this.props;
-      if(!auth.isAuthenticated()) { auth.login(); }
+    let { auth } = this.props;
+    if(!auth.isAuthenticated()) { auth.login(); }
 
-      fetch('/entries/potentials').then(res => {return res.json()}).then((res) => this.setState({potentials: res}));
+    fetch('/entries/potentials')
+      .then(res => {return res.json()})
```

Comment:
do we need `.then(formatForResultList)` here?

Reply:
yes!